### PR TITLE
Update for new dependencies

### PR DESCRIPTION
**Pull request by habolanos**
See [JENKINS-XXXXX](https://issues.jenkins-ci.org/browse/JENKINS-XXXXX).

### Proposed changelog entries

* Entry 1: Issue, Human-readable Text
* ...

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: bom/pom.xml [hunk 1]
```diff
@@ -39,7 +39,7 @@ THE SOFTWARE.
 
 
   <properties>
-    <slf4jVersion>1.7.31</slf4jVersion>
+    <slf4jVersion>1.7.32</slf4jVersion>
```
Comment (basil) [inline bom/pom.xml:42]: Best handled in a separate PR linking to the relevant changelogs (Dependabot will probably open one once we clear the backlog.)

File: bom/pom.xml [hunk 2]
```diff
@@ -66,12 +66,12 @@ THE SOFTWARE.
       <dependency>
         <groupId>org.apache.ant</groupId>
         <artifactId>ant</artifactId>
-        <version>1.10.10</version>
+        <version>1.10.11</version>
```
Comment (basil) [inline bom/pom.xml:69]: Already handled in #5620.

File: bom/pom.xml [hunk 3]
```diff
@@ -66,12 +66,12 @@ THE SOFTWARE.
...
       </dependency>
       <dependency>
         <groupId>commons-io</groupId>
         <artifactId>commons-io</artifactId>
-        <version>2.10.0</version>
+        <version>2.11.0</version>
```
Comment (basil) [inline bom/pom.xml:74]: Already handled in #5623.

File: war/pom.xml [hunk 4]
```diff
@@ -416,7 +416,7 @@ THE SOFTWARE.
       <plugin>
         <groupId>org.eclipse.jetty</groupId>
         <artifactId>jetty-maven-plugin</artifactId>
-        <version>9.4.42.v20210604</version>
+        <version>9.4.43.v20210629</version>
```
Comment (basil) [inline war/pom.xml:419]: Already handled in #5618.

File: pom.xml [hunk 5]
```diff
@@ -100,8 +100,8 @@ THE SOFTWARE.
     <findbugs.excludeFilterFile>${project.basedir}/../src/findbugs/findbugs-excludes.xml</findbugs.excludeFilterFile>
 
     <!-- TODO keep these in sync -->
-    <access-modifier-annotation.version>1.21</access-modifier-annotation.version>
-    <access-modifier-checker.version>1.21</access-modifier-checker.version>
+    <access-modifier-annotation.version>1.22</access-modifier-annotation.version>
+    <access-modifier-checker.version>1.22</access-modifier-checker.version>
```
Comment (basil) [inline pom.xml:104]: Already handled in #5627.

File: pom.xml [hunk 6]
```diff
@@ -100,8 +100,8 @@ THE SOFTWARE.
     <findbugs.excludeFilterFile>${project.basedir}/../src/findbugs/findbugs-excludes.xml</findbugs.excludeFilterFile>
 
     <!-- TODO keep these in sync -->
-    <access-modifier-annotation.version>1.21</access-modifier-annotation.version>
-    <access-modifier-checker.version>1.21</access-modifier-checker.version>
+    <access-modifier-annotation.version>1.22</access-modifier-annotation.version>
```
Comment (basil) [inline pom.xml:103]: Already handled in #5626.

File: bom/pom.xml [hunk 7]
```diff
@@ -400,7 +400,7 @@ THE SOFTWARE.
       <dependency>
         <groupId>jaxen</groupId>
         <artifactId>jaxen</artifactId>
-        <version>1.2.0</version>
+        <version>1.2.0-atlassian-2</version>
```
Comment (basil) [inline bom/pom.xml:403]: Since this isn't an upgrade but rather a switch to a forked version of this dependency, I think we'd need a justification for why we want to switch to the fork. Probably this should be reverted.

File: bom/pom.xml [hunk 8]
```diff
@@ -362,7 +362,7 @@ THE SOFTWARE.
       <dependency><!-- until we get this version through Stapler -->
         <groupId>org.kohsuke.stapler</groupId>
         <artifactId>json-lib</artifactId>
-        <version>2.4-jenkins-2</version>
+        <version>2.4-jenkins-3</version>
```
Comment (basil) [inline bom/pom.xml:365]: Seems legitimate, but since this is a custom fork we should review the changes between the two versions. Best to do this in a separate PR accompanied by a link to the GitHub compare view showing the differences.

File: bom/pom.xml [hunk 9]
```diff
@@ -266,7 +266,7 @@ THE SOFTWARE.
       <dependency><!-- StAX implementation. See JENKINS-2547. -->
         <groupId>org.codehaus.woodstox</groupId>
         <artifactId>wstx-asl</artifactId>
-        <version>3.2.9</version>
+        <version>4.0.0</version>
       </dependency>
```
Comment (basil) [inline bom/pom.xml:270]: This dependency is slated for removal pending the tasks described in [#5604 (comment)](https://github.com/jenkinsci/jenkins/pull/5604#issuecomment-873539241). Most of those tasks are already done, so I'd rather see us take that across the finish line rather than risk breaking something again (the last time I changed this dependency, I caused a regression). Let me know if you'd like to help with these tasks.

**Review by basil — CHANGES_REQUESTED** [review]
My preference is for each dependency update should be handled in a separate PR so that we can carefully review the changes.

File: bom/pom.xml (hunk 8 above)
Comment (daniel-beck) [reply bom/pom.xml:365]: We need to be extremely careful updating this library given how JSON signature validation works. I struggled with unexpectedly broken JSON serialization results a lot in update-center2.

File: bom/pom.xml (hunk 1 above)
Comment (basil) [reply bom/pom.xml:42]: > Dependabot will probably open one once we clear the backlog.

As promised: #5631

**Comment by timja** [comment]
These have all been handled except for json-lib which should be done in a separate PR carefully
